File: MatlabBinaryFiles/readJavaUTFString.m
function [str len] = readJavaUTFString(file);
% function [str len] = readJavaUTFString(file);
% read a string written in Java UTF-8 format from a file.
% The first two bytes are the length of the string, then 
% it's the string. 
len = fread(file, 1, 'int16');
str = char(fread(file, len, 'uchar')');

File: HelperFuncs/readJavaUTFString.R
# Read a Java UTF-8 string. First 2 bytes are length of string, then the string.
# Based on MATLAB code by Michael Oswald

readJavaUTFString <- function(file) {
    # browser()
    len <- pamBinRead(file, 'int16', n=1)
    str <- pamBinRead(file, 'character', n=len)
    # print(len)
    list(len=len, str=str)
}